solcjs

```diff
@@ -52,18 +54,30 @@ function stripBasePath(path) {
     return path;
 }
 
+function toFormattedJson(input) {
+  return JSON.stringify(input, null, program.prettyJson ? 4 : 0);
+}
+
+function reformatJsonIfRequested(inputJson) {
+  return (program.prettyJson ? toFormattedJson(JSON.parse(inputJson)) : inputJson);
+}
+
 var callbacks = undefined
 if (program.basePath)
   callbacks = {'import': readFileCallback};
 
 if (program.standardJson) {
   var input = fs.readFileSync(process.stdin.fd).toString('utf8');
-  var output = solc.compile(input, callbacks);
+  if (program.verbose)
+    console.log('>>> Compiling:\n' + reformatJsonIfRequested(input) + "\n")
+  var output = reformatJsonIfRequested(solc.compile(input, callbacks));
 
   try {
     var inputJSON = smtchecker.handleSMTQueries(JSON.parse(input), JSON.parse(output), smtsolver.smtSolver);
     if (inputJSON) {
-      output = solc.compile(JSON.stringify(inputJSON), callbacks);
+      if (program.verbose)
+        console.log('>>> Retrying compilation with SMT:\n' + toFormattedJson(inputJSON) + "\n")
+      output = reformatJsonIfRequested(solc.compile(JSON.stringify(inputJSON), callbacks));
```

Comment:
@leonardoalt This is a bit off-topic in this PR, but why do we have to retry here? Can't it be detected based on input alone?